api/odl.py

```diff
@@ -1518,3 +1539,38 @@ def _get(self, url, patron=None, headers=None, allowed_response_codes=None):
         self.request_args.append((patron, headers, allowed_response_codes))
         response = self.responses.pop()
         return HTTP._process_response(url, response, allowed_response_codes=allowed_response_codes)
+
+
+class ODLExpiredItemsReaper(IdentifierSweepMonitor):
+    """Responsible for removing expired ODL licenses."""
+
+    SERVICE_NAME = "ODL Expired Items Reaper"
+    PROTOCOL = ODLAPI.NAME
+
+    def __init__(self, _db, collection):
+        super(ODLExpiredItemsReaper, self).__init__(_db, collection)
+
+    def process_item(self, identifier):
+        for licensepool in identifier.licensed_through:
+            licenses_owned = licensepool.licenses_owned
+            licenses_available = licensepool.licenses_available
+
+            for license in licensepool.licenses:
+                if license.is_expired:
+                    licenses_owned -= 1
+                    licenses_available -= 1
```

Comment:
Setting this to -1 instead of 0 ensures that if licenses_owned drops to zero and _then_ the license expires, the expiration is treated as a distinct event. However, I'm not sure there will actually be a real difference in terms of which circulation events are issued.